index.ts

```diff
@@ -5,155 +5,58 @@ import { EventRuleEvent } from "@pulumi/aws/cloudwatch";
+import { AwsServerlessDataPipeline, DataPipelineArgs } from "./serverless/pipeline"
 
 const config = new pulumi.Config();
 const awsConfig = new pulumi.Config("aws")
 const region = awsConfig.require("region");
 const stage = config.require("stage");
 const shards: { [key: string]: number } = config.requireObject("shards");
 
-const dataWarehouseBucket = new aws.s3.Bucket("serverless-db-bucket");
-const athenaResultsBucket = new aws.s3.Bucket("athena-results");
 
-const db = new aws.glue.CatalogDatabase("severless-db", {
-    name: "serverlessdb"
-});
-
-const location = dataWarehouseBucket.arn.apply(a => `s3://${a.split(":::")[1]}`);
-
-const table = new aws.glue.CatalogTable("logs", {
-    name: "logs",
-    databaseName: db.name,
-    tableType: "EXTERNAL_TABLE",
-    storageDescriptor: {
-        location,
-        inputFormat: "org.apache.hadoop.hive.ql.io.parquet.MapredParquetInputFormat",
-        outputFormat: "org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat",
-        serDeInfo: {
-            parameters: { "serialization.format": "1" },
-            name: "ParquetHiveSerDe",
-            serializationLibrary: "org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe"
-        },
-        columns: [
-            {
-                name: "id",
-                type: "string"
-            },
-            {
-                name: "session_id",
-                type: "string"
-            },
-            {
-                name: "message",
-                type: "string"
-            },
-            {
-                name: "event_type",
-                type: "string"
-            }
-        ]
+const columns = [
+    {
+        name: "id",
+        type: "string"
     },
-    partitionKeys: [
-        {
-            name: "inserted_at",
-            type: "string"
-        }
-    ]
-});
-
-const kinesis = new aws.kinesis.Stream("incoming-events", {
-    shardCount: shards[stage],
-});
+    {
+        name: "session_id",
+        type: "string"
+    },
+    {
+        name: "message",
+        type: "string"
+    },
+    {
+        name: "event_type",
+        type: "string"
+    }
+];
 
-let assumeRolePolicy = {
-    "Version": "2012-10-17",
-    "Statement": [
-        {
-            "Action": "sts:AssumeRole",
-            "Principal": {
-                "Service": "firehose.amazonaws.com",
-            },
-            "Effect": "Allow",
-            "Sid": "",
-        },
-    ],
+const dwArgs: DataWarehouseArgs = {
+    columns,
+    tableName: "logs"
 };
 
-const role = new aws.iam.Role("firehoseRose", {
-    assumeRolePolicy: JSON.stringify(assumeRolePolicy),
-});
-
-let kinesisAccess = new aws.iam.RolePolicyAttachment("kinesis-access", {
-    role,
-    policyArn: aws.iam.ManagedPolicies.AmazonKinesisFullAccess,
-});
+const dw = new AwsServerlessDataWarehouse("analytics_dw", dwArgs);
+const {dataWarehouseBucket, queryResultsBucket, database, table} = dw;
```

Comment:
Thoughts on merging theses into a single line statement since dw seems to be used only here.
`const {dataWarehouseBucket, queryResultsBucket, database, table} = new AwsServerlessDataWarehouse("analytics_dw", dwArgs);`